build.rs

```diff
@@ -101,6 +102,43 @@ fn write_postbuild_conf_pri(
     Ok(())
 }
 
+fn get_boost_path() -> Result<String, Box<dyn Error>> {
+    let possible_paths = vec!["/usr/local/include", "/usr/include"];
+    let boost_version = "boost/version.hpp";
+
+    for path in possible_paths {
+        let full_path = Path::new(path).join(boost_version);
+        if full_path.exists() {
+            let file = File::open(full_path)?;
+            let reader = io::BufReader::new(file);
+
+            for line in reader.lines() {
+                match line {
+                    Ok(x) => {
+                        if x.contains("#define BOOST_LIB_VERSION") {
+                            let parts: Vec<&str> = x.split('"').collect();
+                            if parts.len() >= 2 {
+                                let version = parts[1].replace('_', ".");
+                                check_version("boost", &version, 1, 71)?;
+                            } else {
+                                return Err("Error finding boost package verion1"
+                                    .to_string()
```

Comment:
I think you can leave out to_string() on these errors and into() will take care of it